config/python.m4: fix up m4 macro-quoting

jchampio opened the pull request:
We noticed the following error during configure:

    checking Python configuration directory... ./configure: line 9650: 2: command not found

The brackets for the `if` were being swallowed by m4. Double them up to get the correct result for our Bash script.

hlinnaka commented:
Ah, thanks. I've been wondering what's causing that error for a long time!

hlinnaka commented:
Was the script actually working, despite the error? PostgreSQL only seems to have the "else" version of this test. Could we revert this to the way it's in upstream?

jchampio commented:
It was added in b0977a59192, which was trying to explicitly add Python 3 support. @zhangh43, thoughts?

d commented:
:clap: I've been frustrated by the bad m4 in there for a *long* time, thanks for taking the time to fix this. reviewing now

jchampio commented:
> Either use AS_IF([ test "${python_majorversion}" = "2" ] ...)

@d This would have been my preference.

But! Turns out both sides of this `if` generate the same result on my machine, and though I don't expect that to be the case on every platform, I agree with @hlinnaka that it's very unlikely that any platform we're testing even got to the first branch due to this bug. Since no one has been complaining, I've reverted to upstream code here (and added some of the nice debugging messages that upstream has, too).

jchampio commented:
With the latest commit, the Python section of configure looks like this on my machine:
```
checking for python... /usr/local/bin/python
checking for Python distutils module... yes
checking Python configuration directory... /usr/local/opt/python@2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/config
checking Python include directories... -I/usr/local/Cellar/python@2/2.7.15_1/Frameworks/Python.framework/Versions/2.7/include/python2.7
checking how to link an embedded Python application... -F/usr/local/opt/python@2/Frameworks -framework Python
```

jchampio commented:
Thanks for the reviews!
